stumpy/floss.py

```diff
@@ -571,15 +629,30 @@ def update(self, t):
 
         # Ingress
         if self._normalize:
-            T_subseq_isconstant = core.rolling_isconstant(self._T, self._m)
-            M_T, Σ_T = core.compute_mean_std(self._T, self._m)
+            self._T_subseq_isconstant[:-1] = self._T_subseq_isconstant[1:]
+            self._Q_subseq_isconstant = core.rolling_isconstant(
+                self._T[-self._m :], self._m, self._T_subseq_isconstant_func
+            )
+            self._T_subseq_isconstant[-1] = self._Q_subseq_isconstant
+
+            self._M_T[:-1] = self._M_T[1:]
+            self._Σ_T[:-1] = self._Σ_T[1:]
+            self._M_T[-1], self._Σ_T[-1] = core.compute_mean_std(
+                self._T[-self._m :], self._m
+            )
+
             D = core.mass(
                 self._finite_Q,
                 self._finite_T,
-                M_T,
-                Σ_T,
-                T_subseq_isconstant=T_subseq_isconstant,
+                self._M_T,
+                self._Σ_T,
+                T_subseq_isconstant=self._T_subseq_isconstant,
+                Q_subseq_isconstant=self._Q_subseq_isconstant,
             )
+            D[self._Q_subseq_isconstant & self._T_subseq_isconstant] = 0
```

Comment:
Maybe apply the same `mask` here as above for consistency? Or simply add the same comment as above

Reply:
I will. 

---

Now that I am looking at this again, I think we should be able to avoid this post-processing part. Because, `mass` is a public API, and we are already passing these two arguments `Q_subseq_isconstant` and `T_subseq_isconstant` to this function. So, this `mass` function itself is taking care of this. In fact, if we follow the codes, we can see we pass these two params all the way to function `_calculate_squared_distance` (in `core.py`). .